### PR TITLE
Fix some date patterns and add support for more range patterns

### PR DESCRIPTION
**Pull request by chiragjn**
Detailed Changelog

ner_v2.detectors.temporal.date.en.date_detection.DateDetector

- Make year optional in dd/mm/yy (`_gregorian_day_month_year_format`)
- Add support for "-" in dd-mmm-yy format (`_gregorian_day_with_ordinals_month_year_format`)
- Add support for "-" in mmm-dd-yy format (`_gregorian_month_day_with_ordinals_year_format`)
- Make pattern for `_day_month_format_for_arrival_departure` more robust and add range fixing logic
- Make pattern for `_date_range_ddth_of_mmm_to_ddth` more robust and simplify range fixing logic
- Make pattern for `_date_range_ddth_to_ddth_of_next_month` more robust
- Simplify and correct patterns for `_date_identification_everyday_except_weekends` and `_date_identification_everyday_except_weekdays` to get correct original_text
- Fix pattern warning in `_day_range_for_nth_week_month`
- `__get_month_index` and `__get_day_index` now return `int` instead of `str`
- Add various fixme, todo, notes for known issues

chatbot_ner.ner_v2.detectors.temporal.utils

- Add 5 helper methods
    * is_valid_date
    * get_previous_month_number
    * get_next_month_number
    * get_previous_date_with_dd
    * get_next_date_with_dd

ner_v2.detectors.temporal.constant

- MONTH_DICT AND DAY_DICT now have integer keys

ner_v2.detectors.temporal.date.date_detection.DateAdvancedDetector

- Make pattern for `dd th to dd th mmm` more robust
- Make pattern for `dd th mmm to dd th mmm?` more robust
- Two week ranges are only generated now if date type was THIS_DAY or NEXT_DAY
- Change part generation logic in generic range detection
- `_generate_range` renamed to `_fix_day_range`
- Simplify pattern in `_detect_departure_date` and add few variants
- Fix and Simplify patterns in `_detect_return_date` and add few variants
- Remove redundant logic from `_detect_any_date`

### CODE REVIEW

File: ner_v2/detectors/temporal/date/date_detection.py [hunk 1]
```diff
@@ -179,63 +179,68 @@ def _detect_range(self):
...
                     date_dicts[0][temporal_constant.DATE_START_RANGE_PROPERTY] = True
                     date_dicts[1][temporal_constant.DATE_END_RANGE_PROPERTY] = True
                     date_dict_list.extend(date_dicts)
 
-        elif patterns_2:
-            for pattern in patterns_2:
-                date_dicts = self._date_dict_from_text(text=pattern[0])
+        elif dd_mmm_to_dd_matches:
+            for match in dd_mmm_to_dd_matches:
+                print('>>>', match)
```
Comment (srajdev) [inline ner_v2/detectors/temporal/date/date_detection.py]: E1601 print statement used

**Comment by haptik-deployment** [comment]
UNIT TESTS HAVE PASSED... Good To Merge

**Comment by srajdev** [comment]
:+1: No lint errors found.

**Comment by haptik-deployment** [comment]
UNIT TESTS HAVE PASSED... Good To Merge

File: ner_v2/detectors/temporal/date/date_detection.py [hunk 2]
```diff
@@ -322,21 +327,24 @@ def _detect_return_date(self):
...
+        matches_2 = regex_pattern_2.findall(self.processed_text)
+
+        if matches_1:
+            matches = matches_1
+        elif matches_2:
+            matches = matches_2
+
+        matches = matches or []
+        for match in matches:
+            date_dict_list.extend(self._date_dict_from_text(text=match, to_property=True))
```
Comment (krupalmodi18) [inline ner_v2/detectors/temporal/date/date_detection.py]: Did we test this? 
Input to 'self._date_dict_from_text' is changing.

**Comment by srajdev** [comment]
:+1: No lint errors found.

File: ner_v2/detectors/temporal/date/date_detection.py (hunk 2 above)
Comment (chiragjn) [reply ner_v2/detectors/temporal/date/date_detection.py]: Yes this works as now only one groups is selected, match is no longer a tuple

**Comment by haptik-deployment** [comment]
UNIT TESTS HAVE PASSED... Good To Merge

**Pull request merged**